### PR TITLE
Issue 279: Documented stream_iteration pattern

### PR DESCRIPTION
**Pull request by jdgarciauc3m**

### CODE REVIEW

File: doc/stream-iteration.md [hunk 1]
```diff
@@ -1,21 +1,89 @@
...
+Additionally, a filter may have a **Consumer**. The consumer may be any C++
+callable entity that takes values of the result type of the **Transformer**.
 
 ## Details on stream iteration variants
 
-TODO
\ No newline at end of file
+### Stand-alone stream iteration 
+
+A stand alone **stream iteration** has four elements*:
```
Comment (mdolz) [inline doc/stream-iteration.md:68]: Replace `elements*:` by `elements:`.

**Review by mdolz — CHANGES_REQUESTED** [review]
See comments.

**Review by Drioastorga — APPROVED** [review]
Ok.

**Review by javidoc00 — APPROVED** [review]
OK

File: doc/stream-iteration.md (hunk 1 above)
Comment (jdgarciauc3m) [reply doc/stream-iteration.md:68]: Solved.

**Review by mdolz — APPROVED** [review]

**Comment by jdgarciauc3m** [comment]
Closes issue #279

**Pull request merged**